services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/stage2/assign_students.tsx

```diff
@@ -0,0 +1,356 @@
+import React from 'react'
+
+import { useEffect, useState } from 'react'
+
+import ClassroomCard from './classroom_card'
+import CreateAClassInlineForm from './create_a_class_inline_form'
+
+import { canvasProvider, cleverProvider, googleProvider, providerLookup } from '../../../classrooms/providerHelpers'
+import { requestGet } from '../../../../../../modules/request/index'
+import { Snackbar, defaultSnackbarTimeout } from '../../../../../Shared/index'
+import pusherInitializer from '../../../../../../modules/pusherInitializer'
+import ImportProviderClassroomsModal from '../../../classrooms/import_provider_classrooms_modal'
+import LinkProviderAccountModal from '../../../classrooms/link_provider_account_modal'
+import NoClassroomsToImportModal from '../../../classrooms/no_classrooms_to_import_modal'
+import ReauthorizeProviderModal from '../../../classrooms/reauthorize_provider_modal'
+import ButtonLoadingIndicator from '../../../shared/button_loading_indicator'
+
+const canvasIconSrc = `${process.env.CDN_URL}/images/icons/canvas.svg`
+const cleverIconSrc = `${process.env.CDN_URL}/images/icons/clever.svg`
+const googleClassroomIconSrc = `${process.env.CDN_URL}/images/icons/google-classroom.svg`
+const emptyClassSrc = `${process.env.CDN_URL}/images/illustrations/empty-class.svg`
+const smallWhiteCheckSrc = `${process.env.CDN_URL}/images/shared/check-small-white.svg`
+const indeterminateSrc = `${process.env.CDN_URL}/images/icons/indeterminate.svg`
+
+export const createAClassForm = 'createAClassForm'
+
+export const importProviderClassroomsModal = 'importProviderClassroomsModal'
+export const linkCanvasAccountModal = 'linkCanvasAccountModal'
+export const linkCleverAccountModal = 'linkCleverAccountModal'
+export const linkGoogleAccountModal = 'linkGoogleAccountModal'
+export const reauthorizeProviderModal = 'reauthorizeProviderModal'
+export const noClassroomsToImportModal = 'noClassroomsToImportModal'
+
+const AssignStudents = ({
```

Comment:
This refactor is really great. It's exciting to have this big chunk be functional now, and the deduplicated code is readable even though it's more generic. Thanks for that.